### PR TITLE
cleanup build + (additionally) publish distribution-packages via OCM

### PR DESCRIPTION
**Pull request by ccwienk**

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
This PR proposes changes that would break the pipeline definition:
```
cc-utils-build: Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 141, in render
    definition_descriptor = self._render(definition_descriptor)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 185, in _render
    'definition': factory.create_pipeline_definition(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 88, in create_pipeline_definition
    self._apply_traits(variant)
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 177, in _apply_traits
    transformer.process_pipeline_args(pipeline_def)
  File "/usr/lib/python3.12/site-packages/concourse/model/traits/release.py", line 501, in process_pipeline_args
    self.release_step.add_input(
  File "/usr/lib/python3.12/site-packages/concourse/model/step.py", line 426, in add_input
    raise ValueError(f'input already exists: {variable_name}')
ValueError: input already exists: pypi

```

**Comment by ccwienk** [comment]
> This PR proposes changes that would break the pipeline definition:
> 
> ```
> cc-utils-build: Traceback (most recent call last):
>   File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 141, in render
>     definition_descriptor = self._render(definition_descriptor)
>                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 185, in _render
>     'definition': factory.create_pipeline_definition(),
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 88, in create_pipeline_definition
>     self._apply_traits(variant)
>   File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 177, in _apply_traits
>     transformer.process_pipeline_args(pipeline_def)
>   File "/usr/lib/python3.12/site-packages/concourse/model/traits/release.py", line 501, in process_pipeline_args
>     self.release_step.add_input(
>   File "/usr/lib/python3.12/site-packages/concourse/model/step.py", line 426, in add_input
>     raise ValueError(f'input already exists: {variable_name}')
> ValueError: input already exists: pypi
> ```

Thanks @gardener-robot-ci-1 ! I addressed this issue [here](https://github.com/gardener/cc-utils/pull/1004)

File: setup.py [hunk 1]
```diff
@@ -38,10 +34,6 @@ def modules():
 def packages():
     package_names = setuptools.find_packages()
 
-    # remove packages already contained in gardener-cicd-base
-    package_names.remove('ci')
-    package_names.remove('model')
-    package_names.remove('oci')
```
Comment (8R0WNI3) [inline setup.py:46]: I think the `oci` package is/was not actually part of `gardener-cicd-base` but instead it is part of `gardener-oci` (which is still declared as requirement for this package). So, I either suggest to still remove the `oci` package here or, if `gardener-oci` is not required anymore, to also remove it from the build.

**Review by 8R0WNI3 — CHANGES_REQUESTED** [review]

**Review by zkdev — APPROVED** [review]
/lgtm

File: setup.py (hunk 1 above)
Comment (ccwienk) [reply setup.py:46]: you are of course correct - good catch!

**Comment by gardener-robot-ci-2** [comment]
This PR proposes changes that would break the pipeline definition:
```
cc-utils-build: Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 141, in render
    definition_descriptor = self._render(definition_descriptor)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 185, in _render
    'definition': factory.create_pipeline_definition(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 88, in create_pipeline_definition
    self._apply_traits(variant)
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 177, in _apply_traits
    transformer.process_pipeline_args(pipeline_def)
  File "/usr/lib/python3.12/site-packages/concourse/model/traits/release.py", line 501, in process_pipeline_args
    self.release_step.add_input(
  File "/usr/lib/python3.12/site-packages/concourse/model/step.py", line 426, in add_input
    raise ValueError(f'input already exists: {variable_name}')
ValueError: input already exists: pypi

```

**Review by 8R0WNI3 — APPROVED** [review]

**Comment by gardener-robot-ci-2** [comment]
This PR proposes changes that would break the pipeline definition:
```
cc-utils-build: Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 141, in render
    definition_descriptor = self._render(definition_descriptor)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 185, in _render
    'definition': factory.create_pipeline_definition(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 88, in create_pipeline_definition
    self._apply_traits(variant)
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 177, in _apply_traits
    transformer.process_pipeline_args(pipeline_def)
  File "/usr/lib/python3.12/site-packages/concourse/model/traits/release.py", line 501, in process_pipeline_args
    self.release_step.add_input(
  File "/usr/lib/python3.12/site-packages/concourse/model/step.py", line 426, in add_input
    raise ValueError(f'input already exists: {variable_name}')
ValueError: input already exists: pypi

```

**Pull request merged**